server/leafnode_test.go

```diff
@@ -1290,3 +1290,70 @@ func TestLeafNodeExportPermissionsNotForSpecialSubs(t *testing.T) {
 		return nil
 	})
 }
+
+// Make sure that if the node that detects the loop (and sends the error and
+// close the connection) is the accept side, the remote node (the one that solicits)
+// properly use the reconnect delay.
+func TestLeafNodeLoopDetectedOnAcceptSide(t *testing.T) {
+	bo := DefaultOptions()
+	bo.LeafNode.Host = "127.0.0.1"
+	bo.LeafNode.Port = -1
+	b := RunServer(bo)
+	defer b.Shutdown()
+
+	l := &captureErrorLogger{errCh: make(chan string, 10)}
+	b.SetLogger(l, false, false)
+
+	u, _ := url.Parse(fmt.Sprintf("nats://127.0.0.1:%d", bo.LeafNode.Port))
+
+	ao := testDefaultOptionsForGateway("A")
+	ao.Accounts = []*Account{NewAccount("SYS")}
```

Comment:
System account shouldn't be needed for this test

Reply:
It is actually. It is what makes the leafnode connect event be processed and GWs switch to interest-only and therefore LDS subjects to be distributed everywhere (forming the loop).

Reply:
Leafnodes require the cluster (hub) to have a system account and eventing turned on.

Reply:
ok